### DISCUSSION TITLE
Repositrory/list of servers for different platforms (for example C# web api)

### DISCUSSION THREAD

**Discussion by pavelakafoks** [advanced-cropper/vue-advanced-cropper#230] [Ideas]
Hi. I would want to contribute and upload my server side: C# cropper web API.
Do we have repository/folder for that?
Or should I create a my own repository and add a link to it (or may be we have already have list of different server side APIs)?
Thank you.

**Comment by Norserium** [comment]
@pavelakafoks, hello! What do you mean by "C# cropper web API" and how it's related to this library? Did you upload it somewhere now?

**Reply by pavelakafoks** [reply]
There are two options when you crop image:
1) Make some not simple operation to display only selected part of image to final client
2) Change image file physically. I think it is a primary usage case scenario. The library generates request: image url, coordinates and send a request to some back-end service (in my case - C# web api). And server part uploads the image, and crop the needed (selected in the VUE component) part, and save as a new image file. In the future - we will provide this file to final client.

**Reply by pavelakafoks** [reply]
I still didn't upload my solution to my account... If something is already existing - will do PR to this repository/folder.

**Reply by Norserium** [reply]
Well, there are more options.

1. You can crop the image on the client. This library uses `canvas` to implement it.
```js
const { canvas, } = this.$refs.cropper.getResult();
const image = canvas.toDataURL();
```

2. You can crop the image on the server by passing the coordinates to some sort of API.

3. For implementing the preview the specific component ([`Preview`](https://advanced-cropper.github.io/vue-advanced-cropper/components/preview.html#methods)) can be used, that will display "only selected part of image to final client".

I assume, you have created the backend service, that receives URL + coordinates and returns the cropped image to the client. Am I right?

**Reply by pavelakafoks** [reply]
@Norserium , yes, you are absolutely right - I created backed service with you described (point 2). 
I didn't know about option one (canvas.toDataUrl()), but I think it works only to load selected part of image to client.
How do you think - is it useful to have links to repositories, for example in "readme.md" to help users to use this awesome component and do not create server side himself? And how to better implement it? Options:
1) I create and upload to my repo and we add a link to this repo into component documentation
2) like as previous but without a link, who is looking for - can find solution without any links
3) create a folder in this repository and push different API version into it (C#, PHP, Phyton....)

**Reply by Norserium** [reply]
@pavelakafoks, I'm not sure that your library is directly related to this one, I assume that your library can be used with any cropper that allows get the coordinates.

In this circumstances, I suppose, that the best decision write the additional guide that will describe how to crop an image on the server side. This guide may include different links, include the link to your public repository.
